### ISSUE TITLE
Use custom style fails

### ISSUE THREAD

**Issue by RomainQuidet** [mapbox/mapbox-navigation-ios#596]
Init navigation controller with custom styles (custom map url)
Launch navigation

Wanted behavior: custom style (map) applied
Current behavior: mapbox default map is used

Same when styles are set after navigation view controller is visible.

Maybe I missed something in the Style api, but it seems that the custom map url should be used by the map without any other code than setting a style with custom url and giving it to the navigation controller.

**Comment by frederoni** [comment]
Your assumption sounds reasonable. Initializing a `NavigationViewController` with a style and a custom `mapStyleURL` should work as intended.

Can you show a minimal snippet of your implementation?

I just tried this using the latest v0.7.0 which seems to be working.
```swift
let style = DayStyle()
style.mapStyleURL = URL(string: "mapbox://styles/mapbox/satellite-streets-v9")!
let navigationViewController = NavigationViewController(for: route, styles: [style], locationManager: locationManager())
```

You can also create a custom style class:
```swift
class CustomStyle: DayStyle {
    required init() {
        super.init()
        mapStyleURL = URL(string: "mapbox://styles/mapbox/satellite-streets-v9")!
    }
}
```

**Comment by RomainQuidet** [comment]
Hi,
This is exactly what I do, using lib 0.7.0. But I don't get the correct style even with your example.
```swift
let style = DayStyle()
style.mapStyleURL = URL(string: "mapbox://styles/mapbox/satellite-streets-v9")!
let viewController = NavigationViewController(for: route, directions: directions!, styles: [style], locationManager: locationManager)
self.present(viewController, animated: true) {
    print("job done, ride safe")
}
```
I don't get why the applyStyle() does not update the map tiles yet ...

**Comment by RomainQuidet** [comment]
It does not work because there is missing code here...
My carthage file states: github "mapbox/mapbox-navigation-ios" ~> 0.7.0
but when I break into apply() for my custom style, it is called from the NavigationControllerView method:

```swift
    func applyStyle() {
        guard let styles = styles else { return }
        
        for style in styles {
            if style.styleType == styleTypeForTimeOfDay {
                style.apply()
            }
        }
    }
```

as you can see, in the current release branch this method has more code and is actually applying the map styleURL to the map from the style. 
Seems that the 0.7.0 is missing this patch.

**Comment by RomainQuidet** [comment]
This issue has been corrected the 1st September just after 0.7.0... so I don't know how you get it working with release 0.7.0
